providers/basic/basic-auth.go

```diff
@@ -33,23 +32,23 @@ func (c *BasicAuth) Check(r *http.Request) (user string, err error) {
 	user, pass, ok := r.BasicAuth()
 	if !ok {
 		err = core.ErrorAuthenticationFailure
-		return
+		return user, err
```

Comment:
Agree, but you can sometimes find that with a `:=` assignment you can end up defining a new variable that overrides the one from the outer context. Was doing it for clarity of that in this case.